Source/Core/VideoCommon/TextureConverterShaderGen.cpp

```diff
@@ -152,133 +180,73 @@ ShaderCode GenerateTextureConverterShaderCode(APIType api_type,
     switch (uid_data->dst_format)
     {
     case EFBCopyFormat::R4:  // R4
-      colmat[0] = colmat[4] = colmat[8] = colmat[12] = 1;
-      color_mask[0] = 255.0f / 16.0f;
-      color_mask[4] = 1.0f / 15.0f;
+      out.Write("	float red = floor(texcol.r * (255.0 / 16.0)) * (1.0 / 15.0);\n"
```

Comment:
I think it is better to use integer math here, else this might result in an off-by-one again.